src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        addLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) {
+            addLast(value);
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(value);
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addLast(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexPresent(index);
+        Node<T> current = getNode(index);
+        return current.data;
+    }
+
+    private Node<T> getNode(int index) {
+        indexPresent(index);
```

Comment:
indexPresent check is called every time before getNode() method and then inside it
Please choose single place for this check
```suggestion
        indexPresent(index);
```

Reply:
Done